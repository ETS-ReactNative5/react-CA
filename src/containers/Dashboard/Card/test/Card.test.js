import React from 'react'
import { shallow } from 'enzyme'

import ParentCard from '../ParentCard.component'
import ChildCard from '../ChildCard.component'

const componentProps = {
  toggleHeaderTransparency: () => {},
  storeCards: () => {},
  getCardImageByCardId: () => {},
  handleCardLoading: () => {},
  currentUser: { bank_account: { status: null } },
  transactions: [],
  cards: [],
  togglePlaidVisibility: () => {},
  isMobile: false,
}

describe('<ParentCard />', () => {
  it('renders like a beautiful baked apple pie', () => (
    expect(shallow(<ParentCard {...componentProps} />)).toMatchSnapshot()
  ))
})

describe('<ChildCard />', () => {
  it('renders like a Mr.Softee on a sunny NYC day', () => (
    expect(shallow(<ChildCard {...componentProps} />)).toMatchSnapshot()
  ))
})
