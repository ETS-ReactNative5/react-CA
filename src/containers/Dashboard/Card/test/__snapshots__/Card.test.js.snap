// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChildCard /> renders like a Mr.Softee on a sunny NYC day 1`] = `
<div
  className="Card"
>
  <div
    className="noResult"
  >
    No cards
  </div>
</div>
`;

exports[`<ParentCard /> renders like a beautiful baked apple pie 1`] = `
<div
  className="Card"
>
  <div
    className="activateCardWrapper"
  >
    <div
      className="cardContainer"
    >
      <div
        className="activateCard"
      >
        <div
          className="cardTitle"
        >
          Personal Information
        </div>
        <div
          className="cardContent"
        >
          To activate your account we require some personal information.
        </div>
        <Button
          className="activeButton"
          disabled={false}
          icon={null}
          iconOnly={false}
          middle={true}
          nocaps={false}
          onClick={[Function]}
          outline={false}
          primary={true}
          secondary={false}
          small={false}
          text="ACTIVATE"
          textOnly={false}
          tiny={false}
        />
      </div>
    </div>
  </div>
  <div
    className="centerVertical"
  >
    <Button
      className="buttonBig"
      disabled={false}
      icon={null}
      iconOnly={false}
      middle={true}
      nocaps={false}
      onClick={null}
      outline={false}
      primary={false}
      secondary={false}
      small={false}
      text="My Cards"
      textOnly={true}
      tiny={false}
    />
    <div
      align="right"
      className="buttonGroup"
    />
  </div>
  <div
    className="noResult"
  >
    No cards
  </div>
  <Connect(Connect(Connect(LinkPlaidModal)))
    onReload={[Function]}
  />
</div>
`;
